### PR TITLE
Fix IgniteInfo can not import issue

### LINKED ISSUE #8120: cannot import name 'IgniteInfo' from 'monai.config'
```
  File "/opt/toolkit/tutorials/monai/monailabel/apps/radiology/lib/trainers/__init__.py", line 12, in <module>
    from .deepedit import DeepEdit
  File "/opt/toolkit/tutorials/monai/monailabel/apps/radiology/lib/trainers/deepedit.py", line 44, in <module>
    from monailabel.deepedit.handlers import TensorBoardImageHandler
  File "/usr/local/lib/python3.10/dist-packages/monailabel/deepedit/handlers.py", line 16, in <module>
    from monai.config import IgniteInfo
ImportError: cannot import name 'IgniteInfo' from 'monai.config' (/usr/local/lib/python3.10/dist-packages/monai/config/__init__.py)
```

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #8120 .

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by mingxin-zheng** [comment]
Hi @KumoLiu , do you think we'll need to change this line too?

https://github.com/Project-MONAI/MONAI/blob/8546be098daaf6841c39a2748412bbda83929c92/monai/utils/enums.py#L730

The version of Ignite in requirements-dev.txt is 0.4.11

File: monai/config/deviceconfig.py [hunk 1]
```diff
@@ -23,6 +23,8 @@
 import torch
 
 import monai
+from monai.utils.deprecate_utils import deprecated
+from monai.utils.enums import IgniteInfo as NewIgniteInfo
```
Comment (mingxin-zheng) [inline monai/config/deviceconfig.py:27]: I would suggest changing the name of `NewIgniteInfo` to `_IgniteInfo`. Not a blocker though.

File: monai/config/deviceconfig.py [hunk 2]
```diff
@@ -260,5 +262,14 @@ def print_debug_info(file: TextIO = sys.stdout) -> None:
     print_gpu_info(file)
 
 
+@deprecated(since=1.4, removed=1.6, msg_suffix="Please use `monai.utils.enums.IgniteInfo` instead.")
+class IgniteInfo:
+    """
+    IgniteInfo enum is moved to `monai.utils.enums.IgniteInfo`.
```
Comment (mingxin-zheng) [inline monai/config/deviceconfig.py:268]: We can condense these three docstring lines into one:

```
"""Deprecated Import of IgniteInfo enum, which was moved to `monai.utils.enums.IgniteInfo`."""
```

File: monai/config/deviceconfig.py (hunk 1 above)
Comment (KumoLiu) [reply monai/config/deviceconfig.py]: Updated, thanks.

File: monai/config/deviceconfig.py (hunk 2 above)
Comment (KumoLiu) [reply monai/config/deviceconfig.py]: Updated, thx!

**Review by mingxin-zheng — APPROVED** [review]

**Comment by KumoLiu** [comment]
> By the way, do you think a deprecated class should also be included in the `__all__` definitions?
> 
> https://github.com/Project-MONAI/MONAI/blob/754ca155a72e2e448c776d13930f69fce75da909/monai/config/deviceconfig.py#L41

Yes, added.

**Pull request merged**